packages/sources/accuweather/src/endpoint/current-conditions.ts

```diff
@@ -0,0 +1,262 @@
+  currentConditionsResult.windSpeed = Math.trunc(currentConditionsResult.windSpeed * 10)
+
+  // NB: currently all conditions are expected to be numbers
+  Object.entries(currentConditionsResult).forEach(([key, value]) => {
+    if (Number.isNaN(value))
+      throwError(
+        `Attribute '${key}' is not a number: ${JSON.stringify(
+          currentConditionsResult,
+        )}. Current conditions are: ${JSON.stringify(currentConditions)}`,
+      )
+  })
+
+  return currentConditionsResult
+}
+
+export const encodeCurrentConditionsResult = (result: CurrentConditionsResult): string => {
+  const dataTypes = [
+    'tuple(uint256,uint24,uint24,uint24,uint24,int16,uint16,uint16,uint8,uint8,uint8,uint8)',
+  ]
+  const dataValues = [
+    [
+      result.timestamp, // uint256
+      result.precipitationPast12Hours, // uint24
+      result.precipitationPast24Hours, // uint24
+      result.precipitationPastHour, // uint24
+      result.pressure, // uint24
+      result.temperature, // int16
+      result.windDirectionDegrees, // uint16
+      result.windSpeed, // uint16
+      result.precipitationType, // uint8
+      result.relativeHumidity, // uint8
+      result.uvIndex, // uint8
+      result.weatherIcon, // uint8
+    ],
+  ]
+
+  return utils.defaultAbiCoder.encode(dataTypes, dataValues)
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const locationKey = validator.validated.data.locationKey
+  const units = validator.validated.data.units
+  const encodeResult = validator.validated.data.encodeResult ?? true
+
+  validateUnitsParameter(jobRunID, units)
+
+  const url = `currentconditions/v1/${locationKey}.json`
+  const params = {
+    details: true,
+    apikey: config.apiKey,
+  }
+
+  const options = { ...config.api, params, url }
+
+  const response = await Requester.request<Array<CurrentConditions>>(options)
+
+  const currentConditionsList = response.data
+  if (!Array.isArray(currentConditionsList)) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Unexpected response by location key: ${locationKey}. Expected an array but got: ${JSON.stringify(
+        currentConditionsList,
+      )}.`,
+      statusCode: 400,
+    })
+  }
+  let currentConditionsResult: CurrentConditionsResult
+  try {
+    currentConditionsResult = getCurrentConditionsResult(units, currentConditionsList)
+  } catch (error) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Unprocessable response by location key: ${locationKey}. ${error}.`,
+      statusCode: 400,
+    })
+  }
+  let result: CurrentConditionsResult | string
+  if (encodeResult) {
+    try {
+      result = encodeCurrentConditionsResult(currentConditionsResult)
+    } catch (error) {
+      throw new AdapterError({
```

Comment:
same here

Reply:
And all other `AdapterError`s being thrown

Reply:
done!